third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -1397,13 +1418,15 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
             EA::Exp_::Assign(lhs, rhs) => {
                 self.require_impl_language(&loc);
                 let (rhs_ty, rhs) = self.translate_exp_free(rhs);
+                self.translate_lhs = true;
```

Comment:
How about we move this setting and resetting of this flag into `translate_lvalue_list`? Then we won't forget to do it around each of its calls.

Also, do we need to set and reset it around `translate_lvalue_fields`?

Reply:
Actually translate_lvalue_list does not need to be guarded. I updated the use of this flag in the code.